discussion/state.go

```diff
@@ -0,0 +1,54 @@
+package discussion
+
+type State int
```

Comment:
if we are going with `State` as a string instead of int, will that be simpler?
Or is there other concerns?

Reply:
The reason why integer is being used to store the state is to decouple it from the storage and api. State in the API needs to be readable. This is to give more flexibility if we want to name the state differently for each type.